beniget/beniget.py

```diff
@@ -100,21 +126,26 @@ def name(self):
         If the node associated to this Def has a name, returns this name.
         Otherwise returns its type
         """
+        ast = pkg(self.node)
         if isinstance(self.node, (ast.ClassDef,
                                   ast.FunctionDef,
                                   ast.AsyncFunctionDef)):
             return self.node.name
         elif isinstance(self.node, ast.Name):
             return self.node.id
+        elif ast is not gast.gast and isinstance(self.node, ast.arg):
+            return self.node.arg
+        elif ast is not gast.gast and isinstance(self.node, ast.ExceptHandler):
+            return self.node.name
         elif isinstance(self.node, ast.alias):
             base = self.node.name.split(".", 1)[0]
             return self.node.asname or base
-        elif isinstance(self.node, (ast.MatchStar, ast.MatchAs)):
-            if self.node.name:
-                return self.node.name
-        elif isinstance(self.node, ast.MatchMapping):
-            if self.node.rest:
-                return self.node.rest
+        elif _PY310PLUS and isinstance(self.node, (ast.MatchStar, ast.MatchAs)) \
+            and isinstance(self.node.name, str):
+            return self.node.name
+        elif _PY310PLUS and isinstance(self.node, ast.MatchMapping) \
+            and self.node.rest:
+            return self.node.rest
```

Comment:
This is the place I like the less, because we clearly loose the generic aspect of `gast`